UOP1_Project/Assets/Scripts/Sound/SoundManager.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System.Collections;
+using System.Linq;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Audio;
+
+public class SoundManager : MonoBehaviour
+{
+	public static SoundManager singleton;
+
+	[Header ("Init Variables")]
+	[Tooltip ("Amount of sound emitters created on Start")]
+	public int initialPoolSize = 1;
+	[Tooltip ("Frequency at which the sound manager will delete extra emitters that are not in use")]
+	public float trimFrequency = 5;
+	[Tooltip ("Extra time after instanced emitters are not in use anymore before destroying them")]
+	public float extraTimeBeforeTrim = 5;
+
+	[Header ("Mixer Groups")]
+	public AudioMixerGroup masterMixer;
+	public AudioMixerGroup sfxMixer;
+	public AudioMixerGroup musicMixer;
+
+	private List<SoundEmitter> _soundEmitterPool = new List<SoundEmitter>();
+	private GameObject _soundEmittersContainer;
+
+	private void Awake()
+	{
+		DontDestroyOnLoad(this);
+
+		if (!SoundManager.singleton)
+			singleton = this;
+		else
+		{
+			Debug.LogWarning("One sound manager already present in the game, destroying " + this.name + ".");
+			Destroy(this);
+		}
```

Comment:
```suggestion
		if (SoundManager.singleton)
			Debug.LogWarning("One sound manager already present in the game, destroying " + this.name + ".");
			Destroy(this);
		else
		{
			singleton=this;
			DontDestroyOnLoad(this);
		}
```
Two reasons:
1. DontDestroyOnLoad should only be called on the one singleton instance. It's a waste to call it on an object you're about to destroy anyway.
2. I prefer to evaluate the most likely condition first. The reason for this is more apparent with if-elseif-else where there would be multiple conditions to evaluate. By evaluating less likely conditions first, you lose the possibility of an early out. In this case with just an if-else it makes no difference. Purely, just a preference. But thought I'd mention it.

Reply:
The way I see it is, evaluate the condition that would make you not go through the usual process, that way you can remove the else altogether. It looks cleaner, and it's easier to understand that if that condition is met, you're just returning.
``` suggestion
		if (SoundManager.singleton)
              {
			Debug.LogWarning($"One sound manager already present in the game, destroying {name}.");
			Destroy(this);
                      return;
              }

        	singleton = this;
		DontDestroyOnLoad(this);
		
```
(Also added string interpolation)

Reply:
+1 for string interpolation.

Yes, in your suggestion, I believe return is necessary. Destroy does not work immediately, so Awake would continue, singleton would be set, and then destroyed. I think I still prefer explicit else in this case, but works either way.

Reply:
Dave is right, Destroy doesn't work instantly.
Will update the PR with these suggestions during the weekend since all are good changes.